content/tutorials/newsfeed-react.textile

```diff
@@ -0,0 +1,450 @@
+        key: apiKey,
+        clientId: myId,
+        echoMessages: false
+    });
+    const chatChannel = ably.channels.get("chat");
+    const presenceChannel = ably.channels.get("presence");
+    let my = {};
+    my.avatar = avatarsInAssets[getRandomArbitrary(0, 9)];
+    my.name = names[getRandomArbitrary(0, 9)]
+```
+
+Now, let’s make the functional @App@ component and states within it to hold the user input as well as the updates coming in:
+
+```[javascript]
+const App = () => {
+    const [todoInput, setTodoInput] = useState("");
+    const [state, setState] = useState({
+        msgs: [],
+        newMsgs: []
+    })
+    
+    let you = {};
+}
+```
+
+Now, let’s subscribe to the channels in the @useEffect hook@ on mount of the @App@ component:
+
+```[javascript]
+useEffect(() => {
+        
+        chatChannel.subscribe("userInfo", (data) => {
+            var dataObj = JSON.parse(JSON.stringify(data));
+            if (dataObj.clientId !== myId) {
+                you.avatar = dataObj.data.avatar;
+                you.name = dataObj.data.name;
+            }
+        });
+        
+        presenceChannel.presence.subscribe('enter', function (member) {
+            if (member.clientId !== myId) {
+                chatChannel.publish("userInfo", {
+                    "avatar": my.avatar,
+                    "name": my.name
+                });
+            }
+        });
+        presenceChannel.presence.enter();
+        presenceChannel.presence.get(function (err, members) {
+            for (var i in members) {
+                if (members[i].clientId !== myId) {
+                    chatChannel.publish("userInfo", {
+                        "avatar": my.avatar,
+                        "name": my.name
+                    });
+                }
+            }
+        });
+
+        chatChannel.subscribe("chatMessage", (data) => {
+            var dataObj = JSON.parse(JSON.stringify(data));
+            var message = dataObj.data.message;
+            var date = new Date(dataObj.data.date);
+            setState(prevState => {
+                return {
+                  ...prevState,
+                  newMsgs: [...prevState.newMsgs, {"summary": message, "image": you.avatar, "date": date, "name": you.name}]
+                    };
+                });
+        })
+
+      }, []);
+```
+
+As you can see we subscribe to many channels and the publish our information. Let's discuss each of them.
+Using the @presenceChannel@ we mark the presence of the user and also subscribe to @enter@. We also publish the @userInfo@ data to the @chatChannel@.
+Then using the @chatChannel@ we subscribe to the @userInfo@ and get user data.
+Lastly using the @chatChannel@ we subscribe to the @chatMessage@ to get the updates coming in to the newsfeed. We then update the state variables with the new update.
+
+Now, add the following imports statements on top of the file after the @Ably@ import statement to enable use of semantic-ui:
```

Comment:
Don't forget to mention again that it's in the same file (with the file name) that they need to add this.